python/cloudtik/providers/_private/_kubernetes/config.py

```diff
@@ -379,6 +389,23 @@ def validate_kubernetes_config(provider_config: Dict[str, Any], workspace_name:
                            "characters and '-' or '.'".format(provider_config["type"], KUBERNETES_WORKSPACE_NAME_MAX))
 
 
+def prepare_kubernetes_config(config: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
Several things to consider:

1. Can we support:
If user specified his own private key file in config (user already generated externally), we use that. Otherwise, we generate a new pem.

2. Can we move pem file generation process to provider bootstrap? Other providers use bootstrap to generate keys too.